packages/agent/src/hooks/useDidCommPresentationActions.ts

```diff
@@ -46,90 +48,122 @@ export function useDidCommPresentationActions(proofExchangeId: string) {
         name: proofRequest?.name ?? 'Unknown',
       }
 
+      const allCredentialIds = [
+        ...Object.values(anonCredsCredentials.attributes).flatMap((matches) =>
+          matches.map((match) => match.credentialId)
+        ),
+        ...Object.values(anonCredsCredentials.predicates).flatMap((matches) =>
+          matches.map((match) => match.credentialId)
+        ),
+      ]
+      const credentialExchanges = await repository.findByQuery(agent.context, {
+        $or: allCredentialIds.map((credentialId) => ({ credentialIds: [credentialId] })),
+      })
+
+      const attributes = anonCredsCredentials.attributes
       await Promise.all(
-        Object.keys(anonCredsCredentials.attributes).map(async (groupName) => {
+        Object.keys(attributes).map(async (groupName) => {
           const requestedAttribute = proofRequest.requested_attributes[groupName]
           const attributeNames = requestedAttribute?.names ?? [requestedAttribute?.name as string]
-          const attributeArray = anonCredsCredentials.attributes[groupName] as AnonCredsRequestedAttributeMatch[]
-
-          const firstMatch = attributeArray[0]
-
-          if (!firstMatch) {
-            submission.entries.push({
-              credentialName: 'Credential', // TODO: we can extract this from the schema name, but we would have to fetch it
-              isSatisfied: false,
-              name: groupName, // TODO
-              requestedAttributes: attributeNames,
-            })
-          } else {
-            const credentialExchange = await repository.findSingleByQuery(agent.context, {
-              credentialIds: [firstMatch.credentialId],
-            })
-
-            const credentialDisplayMetadata = credentialExchange
-              ? getDidCommCredentialExchangeDisplayMetadata(credentialExchange)
-              : undefined
-
-            submission.entries.push({
-              name: groupName, // TODO: humanize string? Or should we let this out?
-              credentialName: credentialDisplayMetadata?.credentialName ?? 'Credential',
-              isSatisfied: true,
-              issuerName: credentialDisplayMetadata?.issuerName ?? 'Unknown',
-              requestedAttributes: attributeNames,
-            })
-          }
+          const attributeArray = attributes[groupName] as AnonCredsRequestedAttributeMatch[]
+
+          submission.entries.push({
+            inputDescriptorId: groupName,
+            isSatisfied: attributeArray.length >= 1,
+            name: groupName, // TODO
```

Comment:
What is this TODO?

Reply:
The name is the group name if no credentials are found (as that' the only value we have), which can be quite ugly. But this was like this before as well